how-to/applying-a-cowboy-to-qastaging.rst

```diff
@@ -0,0 +1,80 @@
+==============================
+Applying a cowboy to qastaging
+==============================
+
+.. note::
+
+    A cowboy is a hotfix applied to an instance, avoiding a proper deployment
+    by applying code changes directly on an instance.
+
+.. caution::
+
+    A cowboy is a valuable tool to either apply urgent hotfixes or test things
+    quickly e.g. in a staging environment.
+
+    As it avoids a proper deployment, it also comes with a couple of negative
+    side effects:
+
+    - a regular deployment overwrites the cowboy
+
+    - the state of the instance is not clearly defined in the current code
+      and/or deployment configuration
+
+    - and probably some more reasons
+
+    All in all a cowboy should be applied only in rare and extraordinary
+    circumstances.
+
+Before you apply a cowboy, please have a quick conversation with your team
+where the reason for the cowboy is thoroughly discussed.
+
+Applying the cowboy
+===================
+
+#. SSH into bastion ``ssh launchpad-bastion-ps5`` and switch to the
+   ``stb-vbuilder-qastaging`` user by running
+   ``sudo -iu stg-vbuilder-qastaging``.
+
+#. Run ``source ~/.mojorc.qastaging``.
+
+#. Run ``juju config launchpad-appserver build_label`` and take a note of the
+   currently applied git commit id.
+
+#. Checkout that commit id locally.
+
+#. Apply the planned changes to your local checkout and create a local patch
+   file via e.g. ``git diff > <description>.patch``.
+
+#. Log into the appropriate unit where you want to apply the patch to.
+
+    - The following steps describe the process of applying a patch to the
```

Comment:
I'd remove this bullet point and just add these details to the sentence above, example:
```
Log into the appropriate unit where you want to apply the patch to. In this example, we will apply the patch to the application server (``launchpad-appserver``).
```

Reply:
Thanks - that improves readability a lot!